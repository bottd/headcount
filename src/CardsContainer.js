import React from 'react';
import PropTypes from 'prop-types';
import Card from './Card';

const CardsContainer = (props) => {
  const cards = props.stats.map( (place, index) => {
    return (<Card
      removeCompare={props.removeCompare}
      stats={place.stats}
      location={place.location}
      compareCard={props.compareCard}
      key={index}/>);
  });
  return (<div className='Cards'>{cards}</div>);
};

CardsContainer.propTypes = {
  removeCompare: PropTypes.func,
  compareCard: PropTypes.func,
  stats: PropTypes.array
};

export default CardsContainer;
