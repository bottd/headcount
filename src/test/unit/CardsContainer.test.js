import React from 'react';
import { shallow, mount } from 'enzyme';
import CardsContainer from '../../CardsContainer';

describe('CardsContainer', () => {
  it('Renders when no cards are given', () => {
    const wrapper = shallow(<CardsContainer stats={[]}/>);
    expect(wrapper).toMatchSnapshot()
  });

  it('Renders cards given to CardsContainer', () => {
    const card = {
      location: 'COLORADO',
      stats: {
        2004: 0.24,
        2005: 0.278,
        2006: 0.337,
        2007: 0.395,
        2008: 0.536,
        2009: 0.598,
        2010: 0.64,
        2011: 0.672,
        2012: 0.695,
        2013: 0.703,
        2014: 0.741
      }
    }
    const wrapper = shallow(<CardsContainer stats={[card]}/>);
  });

});
