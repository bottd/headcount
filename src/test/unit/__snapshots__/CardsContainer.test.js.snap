// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardsContainer Renders when no cards are given 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardsContainer
    stats={Array []}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [],
      "className": "Cards",
    },
    "ref": null,
    "rendered": Array [],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [],
        "className": "Cards",
      },
      "ref": null,
      "rendered": Array [],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
